onnxruntime/core/providers/cpu/quantization/conv_integer.cc

```diff
@@ -28,12 +32,15 @@ ONNX_OPERATOR_KERNEL_EX(
     10,
     kCpuExecutionProvider,
     KernelDefBuilder()
-        .TypeConstraint("T1", DataTypeImpl::GetTensorType<uint8_t>())
-        .TypeConstraint("T2", DataTypeImpl::GetTensorType<uint8_t>())
+        .TypeConstraint("T1", {DataTypeImpl::GetTensorType<uint8_t>(),
+                               DataTypeImpl::GetTensorType<int8_t>()})
+        .TypeConstraint("T2", {DataTypeImpl::GetTensorType<uint8_t>(),
+                               DataTypeImpl::GetTensorType<int8_t>()})
         .TypeConstraint("T3", DataTypeImpl::GetTensorType<int32_t>()),
     ConvInteger);
 
-Status ConvInteger::Compute(OpKernelContext* context) const {
+template <typename XT, typename WT>
```

Comment:
>emplate <typename XT, typename WT> [](http://example.com/codeflow?start=1&length=34)

This location is fine. No need in header.
I noticed that a large portion of this function is actually type independent.
Would it be possible to split this function in two parts so no code is duplicated unless necessary?
We are very sensitive to binary size increase due to some constrained environments.